### DISCUSSION TITLE
psql syntax

### DISCUSSION THREAD

**Discussion by baoskee** [amacneil/dbmate#228] [Feature Ideas]
What do you think about having psql syntax? It's useful to prompt password for migrating staging and production databases. You don't want the password stored in environment variables. So,

```
dbmate migrate \
      --host=<db_host> \
      --port=<db_port> \
      --username=<db_username> \
      --dbname=<db_name> \
      --password
```

And then it prompts the password for this migration command.

**Comment by amacneil** [comment]
Do you specifically want to pass all the parameters separately using psql syntax, or just the ability to prompt for the password?

Because it would be pretty easy to add a `-p` / `--prompt-password` option (or similar) while keeping the existing URL parameters.

**Reply by baoskee** [reply]
Just password would be great.

**Reply by baoskee** [reply]
Do you have any opinions on the design? I can submit a pull request with simple `-p` appended to any command to insert a password in the URL. It's a little bit awkward since you'd have to parse the URL string and I guess the user would just skip the password in the string?

**Reply by amacneil** [reply]
I think allowing `-p, --password` option in the CLI will be fine.

I think you could implement it in `getDatabaseURL` in main.go somewhere near [here](https://github.com/amacneil/dbmate/blob/main/main.go#L248), so that the rest of dbmate library never needs to know about the password input method. I don't know how exactly to prompt for input (can't be too hard). Ideally with a test case (again, not sure how best to test it but look at common patterns).
